x-pack/plugins/observability/public/hooks/slo/use_create_slo.ts

```diff
@@ -5,75 +5,91 @@
  * 2.0.
  */
 
-import { useCallback, useState } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 import type {
   CreateSLOInput,
   CreateSLOResponse,
+  FindSLOResponse,
   UpdateSLOInput,
   UpdateSLOResponse,
 } from '@kbn/slo-schema';
 
 import { useKibana } from '../../utils/kibana_react';
 
-interface UseCreateOrUpdateSlo {
-  loading: boolean;
-  success: boolean;
-  error: string | undefined;
-  createSlo: (slo: CreateSLOInput) => void;
-  updateSlo: (sloId: string, slo: UpdateSLOInput) => void;
-}
-
-export function useCreateOrUpdateSlo(): UseCreateOrUpdateSlo {
+export function useCreateOrUpdateSlo() {
   const { http } = useKibana().services;
-  const [loading, setLoading] = useState(false);
-  const [success, setSuccess] = useState(false);
-  const [error, setError] = useState<string | undefined>(undefined);
+  const queryClient = useQueryClient();
 
-  const createSlo = useCallback(
-    async (slo: CreateSLOInput) => {
-      setLoading(true);
-      setError('');
-      setSuccess(false);
+  const createSlo = useMutation(
+    ({ slo }: { slo: CreateSLOInput }) => {
       const body = JSON.stringify(slo);
-
-      try {
-        await http.post<CreateSLOResponse>(`/api/observability/slos`, { body });
-        setSuccess(true);
-      } catch (e) {
-        setError(e);
-      } finally {
-        setSuccess(false);
-        setLoading(false);
-      }
+      return http.post<CreateSLOResponse>(`/api/observability/slos`, { body });
     },
-    [http]
+    {
+      mutationKey: ['createSlo'],
+      onSuccess: () => {
+        queryClient.invalidateQueries(['fetchSloList']);
+        queryClient.invalidateQueries(['fetchHistoricalSummary']);
+      },
+    }
   );
 
-  const updateSlo = useCallback(
-    async (sloId: string, slo: UpdateSLOInput) => {
-      setLoading(true);
-      setError('');
-      setSuccess(false);
+  const cloneSlo = useMutation<
```

Comment:
🍰 I wonder if this hook is not doing to many things: create, clone and update. I think it would make sense to split it into 3 hooks: use_clone_slo, use_create_slo, use_update_slo

Reply:
Yeah, that might be wise, as it would mean these mutating hooks will have the same return signature as the delete slo hook. Going to do that!